### ISSUE TITLE
False positive 403 with "yield from" statement 

### ISSUE THREAD

**Issue by pplantinga** [jsh9/pydoclint#127]
Minimal example:

```
def f(args: list):
    """ASDF

    Arguments
    ---------
    args: list
        args

    Yields
    ------
    args: iterable
    """
    yield from args
```

```
    1: DOC403: Function `f` has a "Yields" section in the docstring, but there are no "yield" statements, or the return annotation is not a Generator/Iterator/Iterable. 
    1: DOC404: Function `f` yield type(s) in docstring not consistent with the return annotation. Return annotation does not exist or is not Generator[...]/Iterator[...]/Iterable[...], but docstring "yields" section has 1 type(s).
```

**Comment by jsh9** [comment]
Hi @pplantinga , sorry for the late reply!

The root cause is that `f` lacks a return annotation.  This would produce zero violations:

```python
def f2(args: list) -> Iterator[Iterable]:
    """ASDF

    Arguments
    ---------
    args: list
        args

    Yields
    ------
    args: Iterable
    """
    yield from args
```

I made a code change to make DOC403's error message more self explanatory.  I added this:

> Or it could be because the function lacks a return annotation.
